docker:start always blocks on windows

### Description
Whenever I use docker:start on windows it's always blocking build execution although docker container was successfully started and wait condition satisfied.

I'm using the latest plugin version. I tried explicitly specify docker.follow=false - didn't help. The same config works fine with docker for OSX.

### Info

* d-m-p version : 0.27.1
* Maven version (`mvn -v`) : 3.5.4
* Docker version : 18.06.1-ce-win73
* If it's a bug, how to reproduce :
My docker execution config looks like this:

```
            <plugin>
                <groupId>io.fabric8</groupId>
                <artifactId>docker-maven-plugin</artifactId>
                <version>0.27.1</version>
                <configuration>
                    <images>
                        <image>
                            <name>localstack/localstack:0.8.7</name>
                            <run>
                                <env>
                                    <SERVICES>dynamodb</SERVICES>
                                </env>
                                <ports>
                                    <port>4569:4569</port>
                                </ports>
                                <log/>
                                <wait>
                                    <log>^Ready\.$</log>
                                </wait>
                            </run>
                        </image>
                    </images>
                    <skip>${skipTests}</skip>
                </configuration>
                <executions>
                    <execution>
                        <goals>
                            <goal>start</goal>
                            <goal>stop</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
```
then execution docker:start -X will show that waitcondition was met, execution returned 0 and then process hangs.

I have run Sim4Y sample in debug and I have found that is.available() returns 153, which means that the `is.close()` method is called (which triggers the hanging). With my application, it was returning 0 which means the `is.close()` method was not called.

Because of this, I wonder if we need to care about the `is.available()` value to determine whether or not we should close the InputStream ? Probably not.

Also, here is what I see when I add some logs

```
[INFO] DOCKER> Pattern '^Ready\.$' matched for container f1f8cb78ae5f
f1f8cb> Ready.
*** Calling is.close...
*** Calling request.abort...
```

Then it hangs, so it seems there is a deadlock between these two. 

I tought maybe I can wait for the `finish` method to complete (which is calling `request.abort()`) before calling `is.close()`, which can be done like this:

```
final InputStream is = response.getEntity().getContent();

try {
    while (true) {
        if (!readStreamFrame(is)) {
            return;
        }
    }
} finally {
    // When finish() returns, request == null
    while (request != null) {
        WaitUtil.sleep(500);
    }
    
    if ((is != null)) {
        System.out.println("*** Calling is.close...");
        is.close();
        System.out.println("*** Calling is.close: done.");
    }
}
...
public void finish() {
    if (request != null) {
        System.out.println("*** Calling request.abort...");
        request.abort();
        System.out.println("*** Calling request.abort: done.");
        request = null;
    }
}
```

With this change, it is not hanging anymore. But here is what I see in the logs:
```
*** Calling request.abort...
*** Calling request.abort: done.
*** Calling is.close...
[ERROR] DOCKER> IO Error while requesting logs: java.nio.channels.ClosedChannelException Thread-4
```

Does that mean there is no need to call `is.close()` because it is already? The request.abort() seems to trigger it somehow.

@rohanKanojia, what do you think ? Can you try with these changes on Linux ?

--- a/src/main/java/io/fabric8/maven/docker/access/log/LogRequestor.java
+++ b/src/main/java/io/fabric8/maven/docker/access/log/LogRequestor.java
@@ -30,7 +30,6 @@ import io.fabric8.maven.docker.access.DockerAccessException;
 import io.fabric8.maven.docker.access.UrlBuilder;
 import io.fabric8.maven.docker.access.util.RequestUtil;
 import io.fabric8.maven.docker.util.TimestampFactory;
-import io.fabric8.maven.docker.wait.WaitUtil;
 
 import org.apache.commons.codec.binary.Hex;
 import org.apache.http.HttpResponse;
@@ -184,23 +183,10 @@ public class LogRequestor extends Thread implements LogGetHandle {
         }
 
         final InputStream is = response.getEntity().getContent();
-
-        try {
-            while (true) {
-                if (!readStreamFrame(is)) {
-                    return;
-                }
-            }
-        } finally {
-            // When finish() returns, request == null
-            while (request != null) {
-                WaitUtil.sleep(500);
-            }
-            
-            if ((is != null)) {
-                System.out.println("*** Calling is.close...");
-                is.close();
-                System.out.println("*** Calling is.close: done.");
+       
+        while (true) {
+            if (!readStreamFrame(is)) {
+                return;
             }
         }
     }
@@ -224,9 +210,7 @@ public class LogRequestor extends Thread implements LogGetHandle {
     @Override
     public void finish() {
         if (request != null) {
-            System.out.println("*** Calling request.abort...");
             request.abort();
-            System.out.println("*** Calling request.abort: done.");
             request = null;
         }
     }

--- a/src/main/java/io/fabric8/maven/docker/access/log/LogRequestor.java
+++ b/src/main/java/io/fabric8/maven/docker/access/log/LogRequestor.java
@@ -30,6 +30,8 @@ import io.fabric8.maven.docker.access.DockerAccessException;
 import io.fabric8.maven.docker.access.UrlBuilder;
 import io.fabric8.maven.docker.access.util.RequestUtil;
 import io.fabric8.maven.docker.util.TimestampFactory;
+import io.fabric8.maven.docker.wait.WaitUtil;
+
 import org.apache.commons.codec.binary.Hex;
 import org.apache.http.HttpResponse;
 import org.apache.http.StatusLine;
@@ -190,8 +192,15 @@ public class LogRequestor extends Thread implements LogGetHandle {
                 }
             }
         } finally {
-            if ((is != null) && (is.available() > 0)) {
+            // When finish() returns, request == null
+            while (request != null) {
+                WaitUtil.sleep(500);
+            }
+            
+            if ((is != null)) {
+                System.out.println("*** Calling is.close...");
                 is.close();
+                System.out.println("*** Calling is.close: done.");
             }
         }
     }
@@ -215,7 +224,9 @@ public class LogRequestor extends Thread implements LogGetHandle {
     @Override
     public void finish() {
         if (request != null) {
+            System.out.println("*** Calling request.abort...");
             request.abort();
+            System.out.println("*** Calling request.abort: done.");
             request = null;
         }
     }

